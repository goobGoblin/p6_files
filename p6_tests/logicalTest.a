logicalTest : () -> bool {
  a : bool;
  b : bool;
  c : bool;
  
  a = true;
  b = false;
  
  c = !a;
  a = a and b;
  b = a or c;
  
  return b;
}
