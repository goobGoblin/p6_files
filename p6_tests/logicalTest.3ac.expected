[BEGIN GLOBALS]
logicalTest
[END GLOBALS]
[BEGIN <init> LOCALS]
[END <init> LOCALS]
fun_<init>: enter <init>
lbl_0:      leave <init>
[BEGIN logicalTest LOCALS]
a (local var of 8 bytes)
b (local var of 8 bytes)
c (local var of 8 bytes)
tmp0 (tmp var of 8 bytes)
tmp1 (tmp var of 8 bytes)
tmp2 (tmp var of 8 bytes)
[END logicalTest LOCALS]
fun_logicalTest: enter logicalTest
            [a] := 1
            [b] := 0
            [tmp0] := NOT64 [a]
            [c] := [tmp0]
            [tmp1] := [a] AND64 [b]
            [a] := [tmp1]
            [tmp2] := [a] OR64 [c]
            [b] := [tmp2]
            setret [b]
            goto lbl_1
lbl_1:      leave logicalTest
